File: business-context/src/main/java/cz/filipklimes/diploma/framework/businessContext/expression/logical/Or.java
package cz.filipklimes.diploma.framework.businessContext.expression.logical;

import cz.filipklimes.diploma.framework.businessContext.BusinessContext;
import cz.filipklimes.diploma.framework.businessContext.expression.BinaryOperator;
import cz.filipklimes.diploma.framework.businessContext.expression.Expression;

import java.util.*;

public final class Or extends BinaryOperator<Boolean, Boolean, Boolean>
{

    public Or(final Expression<Boolean> left, final Expression<Boolean> right)
    {
        super(left, right);
    }

    @Override
    public Boolean interpret(final BusinessContext context)
    {
        return getLeft().interpret(context) || getRight().interpret(context);
    }

    @Override
    public Map<String, String> getProperties()
    {
        return Collections.emptyMap();
    }

    @Override
    public String getName()
    {
        return "logical-or";
    }

    @Override
    public String toString()
    {
        return String.format("%s or %s", getLeft(), getRight());
    }

}

File: nodejs/business-context/src/expression/logical/Or.js
export default class Or {
    constructor(left, right) {
        this.left = left
        this.right = right
    }

    interpret(context) {
        return this.left.interpret(context) || this.right.interpret(context)
    }

    getName() {
        return 'logical-or'
    }

    getArguments() {
        return [this.left, this.right]
    }

    getProperties() {
        return {}
    }

    toString() {
        return this.left.toString() + " or " + this.right.toString()
    }

    clone() {
        return new Or(
            this.left.clone(),
            this.right.clone(),
        )
    }
}
